openapi-v1.yaml

```diff
@@ -6036,3 +6125,128 @@ paths:
           description: error response
           schema:
             $ref: "#/definitions/Error"
+
+  #
+  # Groups: Swagger does not allow optional path parameters so we have to define two endpoints
+  # /groups/{namespace} for the root group
+  # /groups/{namespace}/{id} for other groups
+  # This will add additional methods for root group like listGroup and listRootGroup
+  # To make UI clients simpler, in the backend we will allow requests with empty id
+  # and reroute them to the root group
+  #
+
+  /groups/{namespace}/{id}:
```

Comment:
Just like arrays, and other assets the user should always reference by `name` not uuid of a group. Lets keep the route parameter name similar.

Reply:
Two groups can have the same name. Should we go for `/groups/namespace/{type}/{path}` endpoints?